### ISSUE TITLE
Record2Vec: Unsupervised Representation Learning for Structured Records

### ISSUE THREAD

**Issue by wwacky** [wwacky/survey#3]
## 一言でいうと
分散表現を構造化されたデータレコードの属性値に適用する。これによってクラスタリングやrecord matching(名寄せ？)、分類モデルに適用することが出来る。
TF-IDFよりも蜜ベクトルで、教師ラベルも不要な汎用的な手法であるというのが筆者らの主張。

### 論文リンク
[https://www.aboutamazon.com](https://d39w7f4ix9f5s9.cloudfront.net/8b/ad/fa5173cc4600b0093933e932fc75/sim-record2vecicdmshort2018-1.pdf)

### 著者/所属機関
- Adelene Y.L. Sim (Amazon.com)
- Andrew Borthwick (Amazon.com)

### 投稿日付
2018-11-17 ICDM2018

## 概要

- 商品タイトルのattributeが“iPhone 8, 4.7, 64 GB,Black”であれば、ブランドのattributeが"Apple"と推測できる。これをembeddingから学習することが出来るようにする
- {Documents, Sentence, Words}という構成を、{Records, Attributes, Tokens}という構造化データにあてはめて分散表現を構築する。DocumentsとRecords、SentenceとAttributes、WordsとTokensが対応している

- 利用データ：descriptive text, categorical, ordinal, single word/token and even numerical

## 新規性・差分
- 構造化されたデータに対する分散表現の適用
- 教師ラベルがなくても学習が行える汎用性

## 手法
Recordには複数のAttributeがあり、それらの値をtokenとする。
同一Recordに出現する各AttributeのTokenを埋め込んでベクトル表現する。
損失関数はtriplet loss functionを利用し、Negative Samplingは別レコードの同一AttributeのToken(ただしAttributeのTokenが違うという制約はつける)で行い学習させる。
埋め込みはCBowとSkip-Gramの両方を試す。埋め込み後のベクトルは50次元。50epochsで学習。
ノイズを減らすために、単語は頻出の200単語にする？

![image](https://user-images.githubusercontent.com/4422849/53172066-c2b86680-3627-11e9-8aeb-9d4c682c7186.png)
※Record2Vec: Unsupervised Representation Learning for Structured Recordsより引用

たぶんイメージとしてはこんな感じ。
Attributeの値が複数Tokenの場合はベクトル平均なので、商品タイトルや商品説明はベクトルの平均値になるはず。
![image](https://user-images.githubusercontent.com/4422849/53212225-80bf0d00-3687-11e9-84a1-48daad3b541a.png)
※この図は自分で作成

## 結果
「クラスタリング」「record matching task (名寄せ？)」「映画のratingとジャンル予測」で評価。
線形分類(ロジスティックか線形回帰)、非線形分類はXGBoost。

データの内容
![image](https://user-images.githubusercontent.com/4422849/53216730-e8318880-3698-11e9-9ff9-1f6f3c67c43d.png)
※Record2Vec: Unsupervised Representation Learning for Structured Recordsより引用

- クラスタリングと名寄のタスクでの評価

k-meansで100クラスタに分類。データに正解クラスタIDを振って評価？

- record matching task

重複レコードを見つけるタスク。
レコードを50次元のベクトルに埋め込んでおく。
2つのレコードが重複かどうかは、2つの50次元のベクトルのコサイン類似度を算出し、類似度を素性として判別モデルを適用する（LinearとXGBoostの両方）。

評価用データの内容
Cora：Cora Computer Science Research Paper Engineに手でラベルを付けた出版物。
DBLP-ACM：サイエンス系の出版物。record matching task(名寄せ？)のためにmatch labelがある
DBLP-Scholar：サイエンス系の出版物。Google Schlarから取れるもの。DBLP-ACMと同じ様な形式
Abt-Buy：Abt.comとBuy.comの商品データをマージしたもの
Amazon-Google：Amazonの商品データに対してGoogleで取得したUniversal Product Codeを使って結びつける

TF-IDF-SVDよりは良いが、State of the artより劣る。筆者らとしてはState of the artの手法は文字列マッチングを用いた手法なので直接比較できないということらしい。

![image](https://user-images.githubusercontent.com/4422849/53214609-8cfb9800-3690-11e9-9742-02f4dca40b99.png)
※Record2Vec: Unsupervised Representation Learning for Structured Recordsより引用

- 映画のratingとジャンル予測

IMDbのデータセットを利用。
線形モデルだと提案手法が最もMean squared errorが小さいが、XGBoostだとTF-IDF-SVDの方が良くなっている。

## コメント

record matching taskではState of the artの方が性能が良いので、以下の論文も読んだほうがよいかも。

- [10] A. Bordes, X. Glorot, J. Weston, and Y. Bengio, “A semantic matching energy function for learning with multi-relational data,” Machine Learning, vol. 94, no. 2, pp. 233–259, Feb 2014.
- [18] H. Ko ̈pcke, A. Thor, and E. Rahm, “Evaluation of entity resolution approaches on real-world match problems,” Proc. VLDB Endow., vol. 3, no. 1-2, pp. 484–493, Sep. 2010.

## 略語
頭字語 | 意味
--- | ---
SG | skip-gram
SOTA | State-of-the-art
R2V | Record2Vec
